language/stdlib/modules/libra_account.mvir

```diff
@@ -57,22 +76,93 @@ module LibraAccount {
         payer: address,
     }
 
+    // Events
+    // Derive a fresh unique id by using sender's HandleIdGenerator. The generated bytearray is indeed unique because it
+    // was derived from the hash(sender's HandleIdGenerator || sender_address). This module guarantees that the
+    // HandleIdGenerator is only going to be monotonically increased and there's no way to revert it or destroy it. Thus
+    // such counter is going to give distinct value for each of the new event stream under each sender. And since we
+    // hash it with the sender's address, the result is guaranteed to be globally unique.
+    fresh_guid(counter: &mut Self.HandleIdGenerator): bytearray{
+        let count: &mut u64;
+        let sender_bytes: bytearray;
+        let count_bytes: bytearray;
+        let preimage: bytearray;
+
+        count = &mut move(counter).counter;
+
+        sender_bytes = AddressUtil.address_to_bytes(get_txn_sender());
```

Comment:
I think Ben is right; this PR does not fix the problem. I think we will need to replace the  `create_account` bytecode (as discussed with @dariorussi and others) before we can actually address this as.

Reply:
> > I wonder if it's just worth making an API that meets use case 2 in a single round trip. For example a call to get recent events might pass the access path for the sent_events event handle in account A. The API call would return the following information:
> > (1) From the state tree, the entire EventHandle. This would allow verification of the most recent counter value (used to authenticate the returned events) as well as the GUID of the event
> > (2) The specific events matching the query
> 
> This is almost exactly how the events API works [today](https://github.com/libra/libra/blob/master/types/src/proto/get_with_proof.proto#L261-294). The only difference is that it doesn't return the handle separately because there's no need; each event will already contain it.
> 
> We actually do not have a "query-by-event-handle" API, but could choose to have one in the future.

Yea the current API look like following: Give me the Sent/Recv events for account address X with some ranges of sequence number. In the node, it was implemented by first issuing a query to the StateStore to fetch the `EventHandle` for SentEvent/ReceivedEvent and use the guid in the handle to get the entry to the Event store, and get the events from there